src/Data/URI.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace ILIAS\Data;
+
+
+/**
+ * The scope of this class is split ilias-conform URI's into components.
+ * Please refer to RFC 3986 for details.
+ * Notice: ilias-confor URI's will form a SUBSET of RFC 3986.
+ * We limit the set of allowed sub-delimiters in authority.
+ * We require a schema and an authority to be present.
+ * If any part is located and it is invalid an exception will be thrown
+ * instead of just omiting it.
+ */
+class URI
+{
+	protected $schema;
+	protected $authority;
+	protected $path;
+	protected $query;
+	protected $fragment;
+
+
+	const ALPHA_SUB_REGEX = 'A-Za-z';
+	const DIGIT_SUB_REGEX = '0-9';
+	const ALPHA_DIGIT_SUB_REGEX = self::ALPHA_SUB_REGEX.self::DIGIT_SUB_REGEX;
+	const UNRESERVED_SUB_REGEX = self::ALPHA_DIGIT_SUB_REGEX.'\\-\\._~';
```

Comment:
Diese Backslashes machen mich verrückt und ich glaube wir brauchen die auch nicht (?) weil du das später eh in `[]` einfügst, oder?